erpnext/hr/doctype/employee_checkin/employee_checkin.py

```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2018, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import datetime, time
+import frappe
+from frappe.model.document import Document
+
+class EmployeeCheckin(Document):
+	def validate(self):
+		if self.employee:
+			attendance = frappe.db.get_list("Attendance",
+				{"employee": self.employee,
+				"attendance_date": datetime.datetime.now()},
+				["name"]
+			)
+			if not attendance:
+				employee_doc = frappe.get_doc("Employee", self.employee)
+				## Create Employee Attendance
+				employee_attendance = frappe.new_doc("Attendance")
+				employee_attendance.company = employee_doc.company
+				employee_attendance.employee = employee_doc.name
+				employee_attendance.employee_name = employee_doc.employee_name
+				employee_attendance.status = "Present"
+				employee_attendance.attendance_date =  datetime.datetime.now()
+				employee_attendance.department = employee_doc.department
+				employee_attendance.save()
+				employee_attendance.submit()
+
+		if self.out_time:
+			in_time = datetime.datetime(*time.strptime(str(self.in_time), "%Y-%m-%d %H:%M:%S")[:6])
+			out_time = datetime.datetime(*time.strptime(str(self.out_time), "%Y-%m-%d %H:%M:%S")[:6])
+			if in_time > out_time:
+				frappe.throw("Out-time must be later than In-time.")
+			else:
+				duration = out_time - in_time
+				self.duration = duration
+
+@frappe.whitelist(allow_guest=False)
+def punch_in(rfid_tag):
+	if "Auto Attendance" in frappe.get_roles(frappe.session.user):
```

Comment:
Should be based on a checkbox (Allow Checkin?) in Employee. Remove hard coded role.
https://github.com/frappe/erpnext/wiki/Model-Design-Guidelines#avoid-hard-coding-of-roles